testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDatabricksWithGlueMetastoreCleanUp.java

```diff
@@ -99,9 +101,13 @@ private void cleanSchema(String schema, long startTime, AWSGlueAsync glueClient)
         }
         log.info("Dropped %d tables in schema %s", droppedTablesCount, schema);
         if (!schema.equals("default") && findAllTablesInSchema(schema).isEmpty()) {
+            Database database = glueClient.getDatabase(new GetDatabaseRequest().withName(schema)).getDatabase();
+            Instant createTime = database.getCreateTime().toInstant();
             try {
-                onTrino().executeQuery("DROP SCHEMA IF EXISTS " + schema);
-                log.info("Dropped schema %s", schema);
+                if (createTime.isBefore(SCHEMA_CLEANUP_THRESHOLD)) {
```

Comment:
If we are not going to delete the schema, we should also abstain from deleting tables in that schema